src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -943,6 +952,34 @@ public void pasteEntry() {
             return;
         }
 
+        Optional<String> optionalBibFolderPath = copiedBibDatabaseContext.getMetaData().getDefaultFileDirectory();
+        if(optionalBibFolderPath.isEmpty()){
+            LOGGER.warn("can not find the file path");
+            return;
+        }
+
+        Path sourceBibFolder = Paths.get(optionalBibFolderPath.get());
+
+        for(BibEntry entry:entriesToAdd){
+            List<LinkedFile> linkedFiles = entry.getFiles();
+            for(LinkedFile linkedFile: linkedFiles){
+                if("pdf".equalsIgnoreCase(linkedFile.getFileType())){
+                    List<Path> folderList = List.of(sourceBibFolder);
+                    Optional<Path> sourcePath = linkedFile.findIn(folderList);
+                    if(sourcePath.isPresent() && Files.exists(sourcePath.get())){
+                        Path targetFolder = bibDatabaseContext.getMetaData().getDefaultFileDirectory().map(Paths::get).orElseThrow();
+                        Path targetPath = targetFolder.resolve(sourcePath.get().getFileName());
+                        try{
+                            Files.copy(sourcePath.get(), targetPath, StandardCopyOption.REPLACE_EXISTING);
+                        } catch (IOException e) {
+                            LOGGER.error("failed to copy PDF file: {}", sourcePath.get(),e);
+                        }
+                    }else{
+                        LOGGER.warn("did not find PDF: {}", linkedFile.getLink());
+                    }
```

Comment:
No manual moving. Try to adapt `  public boolean copyOrMoveToDefaultDirectory(boolean shouldMove, boolean shouldRenameToFilenamePattern) throws IOException {` from `LinkedFileHandler.java`.

I think, it can be used if user-specific file directory exists.

Proper code for that is to introduce `FileDirectoriesInfo`. (see below)

org.jabref.model.database.BibDatabaseContext#getFirstExistingFileDir needs to be made for FileDirectoriesInfo.

Then, the result can be checked if it is NOT mainFileDirectory.

Then, before you even start copy, check if the file is reachable from the NEW library (org.jabref.model.entry.LinkedFile#findIn(java.util.List<java.nio.file.Path>)). If yes, do not copy. 

Handle all files, not just PDFs.

Introduced following in `BibdatabaseContext`:

```java
    public record FileDirectoriesInfo(
            Optional<Path> mainFileDirectory,
            Optional<Path> librarySpecificDirectory,
            Optional<Path> userFileDirectory) {
    }

    public FileDirectoriesInfo getFileDirectoriesInfo(FilePreferences preferences) {
        Optional<Path> mainFileDirectory = metaData.getUserFileDirectory(preferences.getUserAndHost()).map(dir -> getFileDirectoryPath(dir));
       if (mainFileDirectory.isEmpty() && preferences.shouldStoreFilesRelativeToBibFile()) {
             // TODO
       }

        Optional<Path> librarySpecificDirectory = metaData.getLibrarySpecificFileDirectory()
                                                          .map(Path::of);

        Optional<Path> userFileDirectory = metaData.getUserFileDirectory(preferences.getUserAndHost())
                                                   .map(Path::of);

        return new FileDirectoriesInfo(mainFileDirectory, librarySpecificDirectory, userFileDirectory);
    }
```

Replace the TODO part with something like follows - you need to extract that part from org.jabref.model.database.BibDatabaseContext#getFileDirectories into a new method.

```
            getDatabasePath().ifPresent(dbPath -> {
                Path parentPath = dbPath.getParent();
                if (parentPath == null) {
                    parentPath = Path.of(System.getProperty("user.dir"));
                    LOGGER.warn("Parent path of database file {} is null. Falling back to {}.", dbPath, parentPath);
                }
                Objects.requireNonNull(parentPath, "BibTeX database parent path is null");
                fileDirs.add(parentPath.toAbsolutePath());
            });

```


---